### ISSUE TITLE
What is the difference between the DOH POST request sent by DOG and by Q?

### ISSUE THREAD

**Issue by fxzxmic** [natesales/q#76]
I created a simple DOH forwarding myself, but found that using DOG and Q's GET mode worked fine, but using Q's POST mode would return a 500 error. What is the reason for this?

**Comment by fxzxmic** [comment]
I found the reason: the way Q sends POST requests is not accepted by mainstream DOH servers.
```
$ q @https://cloudflare-dns.com/dns-query github.com A --http-method=POST
FATA[0000] got status code 415 from https://cloudflare-dns.com:443/dns-query?dns=oXsBAAABAAAAAAAABmdpdGh1YgNjb20AAAEAAQ 
$ q @https://dns.google/dns-query github.com A --http-method=POST
FATA[0000] got status code 400 from https://dns.google:443/dns-query?dns=APcBAAABAAAAAAAABmdpdGh1YgNjb20AAAEAAQ
```

**Comment by natesales** [comment]
Indeed! Fixed in v0.16.0

```
~ ▴ q -V
https://github.com/natesales/q version 0.16.0 (7feffa12d62cbe9dd0fc71138c54daebd7b6748e 2023-12-13T04:54:44Z)
~ ▴ q @https://cloudflare-dns.com/dns-query github.com A --http-method=POST
github.com. 15s A 140.82.114.4
~ ▴ q @https://dns.google/dns-query github.com A --http-method=POST
github.com. 1m A 140.82.112.4
```
